Mods/SML/Source/SML/Private/Patching/BlueprintHookManager.cpp

```diff
@@ -84,53 +75,128 @@ void UBlueprintHookManager::InstallBlueprintHook(UFunction* Function, int32 Hook
-
-	//Insert jump to original location for running code after hook
+	//Insert jump to after the resolved hook offset location for running code after hook
 	AppendedCode.Add(EX_Jump);
-	const int32 JumpDestination = HookOffset + BytesAvailable;
+	const int32 JumpDestination = ResolvedHookOffset + UBlueprintHookManager::JumpBytesRequired;
 	WRITE_UNALIGNED(AppendedCode, CodeSkipSizeType, JumpDestination);
 
-	//Finish generated code with EX_EndOfScript to avoid any surprises
-	AppendedCode.Add(EX_EndOfScript);
-
-
 	//Append generated code to the end of the function's original code now
 	const int32 StartOfAppendedCode = OriginalCode.Num();
 	OriginalCode.Append(AppendedCode);
 
-	//Fill space with EX_EndOfScript before replacement for safety
-	FPlatformMemory::Memset(&OriginalCode[HookOffset], EX_EndOfScript, BytesAvailable);
-
 	//Actually insert jump to the start of appended code to original hook location
-	OriginalCode[HookOffset] = EX_Jump;
-	FPlatformMemory::WriteUnaligned<CodeSkipSizeType>(&OriginalCode[HookOffset + 1], StartOfAppendedCode);
+	OriginalCode[ResolvedHookOffset] = EX_Jump;
+	FPlatformMemory::WriteUnaligned<CodeSkipSizeType>(&OriginalCode[ResolvedHookOffset + 1], StartOfAppendedCode);
 
 #if DEBUG_BLUEPRINT_HOOKING
-	DebugDumpFunctionScriptCode(Function, HookOffset, TEXT("AfterHook"));
+	DebugDumpFunctionScriptCode(Function, RequestedHookOffset, ResolvedHookOffset, TEXT("AfterHook"));
 #endif
 }
 
-int32 UBlueprintHookManager::PreProcessHookOffset(UFunction* Function, int32 HookOffset) {
-	if (HookOffset == EPredefinedHookOffset::Return) {
-		//For now Kismet Compiler will always generate only one Return node, so all
-		//execution paths will end up either with executing it directly or jumping to it
-		//So we need to hook only in one place to handle all possible execution paths
-		FSMLKismetBytecodeDisassembler Disassembler;
-		int32 ReturnOffset;
-		const bool bIsValid = Disassembler.FindFirstStatementOfType(Function, 0, EX_Return, ReturnOffset);
-		checkf(bIsValid, TEXT("EX_Return not found for function %s"), *Function->GetPathName());
-		return ReturnOffset;
+void UBlueprintHookManager::ModifyJumpTargetsForNewHookOffset(TArray<uint8>& Script, TSharedPtr<FJsonObject> Expression, int32 HookOffset)
+{
+	double OpcodeDouble;
+	if (!Expression->TryGetNumberField(TEXT("Opcode"), OpcodeDouble)) {
+		// No opcode means it's not an instruction, so we can just return;
+		return;
+	}
+	int32 Opcode = (int32)OpcodeDouble;
+
+	//A computed jump could do anything at runtime - it could jump to before or after the hook, so we have no way of knowing how
+	//it needs to be modified to work correctly. For now, the only predictable solution is to forbid hooking functions with them.
+	checkf(Opcode != EX_ComputedJump, TEXT("Cannot hook a blueprint function that contains an EX_ComputedJump instruction. There's no way to guarantee it would not crash at some point."));
```

Comment:
As per your PR's description, `checkf` does nothing at runtime

Reply:
Good catch, fixed. I also fixed a few other checks in the file.

Reply:
Ack